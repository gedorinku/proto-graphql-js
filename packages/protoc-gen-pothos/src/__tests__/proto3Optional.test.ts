import { itGeneratesDSLsToMatchSnapshtos } from "./__helpers__/process.test.helper";

describe("proto3 optional", () => {
  itGeneratesDSLsToMatchSnapshtos("testapis.proto3_optional", [
    "proto3_optional/proto3_optional.pb.pothos.ts",
  ]);
});
