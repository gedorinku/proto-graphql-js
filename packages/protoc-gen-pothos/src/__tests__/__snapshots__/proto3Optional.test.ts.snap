// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`proto3 optional with ts-proto generates pothos DSLs: proto3_optional/proto3_optional.pb.pothos.ts 1`] = `
"// Code generated by protoc-gen-pothos. DO NOT EDIT.
// source: testapis/proto3_optional/proto3_optional.proto

/* eslint-disable */

import { InputObjectRef } from "@pothos/core";
import { builder } from "../../builder";
import { Message } from "./testapis/proto3_optional/proto3_optional";

export const Message$Ref = builder.objectRef<Message>("Message");
builder.objectType(Message$Ref, {
  name: "Message",
  fields: (t) => ({
    requiredStringValue: t.expose("requiredStringValue", {
      type: "String",
      nullable: false,
      extensions: {
        protobufField: {
          name: "required_string_value",
          typeFullName: "string",
        },
      },
    }),
    optionalStringValue: t.expose("optionalStringValue", {
      type: "String",
      nullable: true,
      extensions: {
        protobufField: {
          name: "optional_string_value",
          typeFullName: "string",
        },
      },
    }),
  }),
  isTypeOf: (source) => {
    return (source as Message | { $type: string & {}; }).$type ===
      "testapis.proto3_optional.Message";
  },
  extensions: {
    protobufMessage: {
      fullName: "testapis.proto3_optional.Message",
      name: "Message",
      package: "testapis.proto3_optional",
    },
  },
});

export type MessageInput$Shape = {
  requiredStringValue: Message["requiredStringValue"];
  optionalStringValue?: Message["optionalStringValue"] | null;
};

export const MessageInput$Ref: InputObjectRef<MessageInput$Shape> = builder
  .inputRef<MessageInput$Shape>("MessageInput").implement({
    fields: (t) => ({
      requiredStringValue: t.field({
        type: "String",
        required: true,
        extensions: {
          protobufField: {
            name: "required_string_value",
            typeFullName: "string",
          },
        },
      }),
      optionalStringValue: t.field({
        type: "String",
        required: false,
        extensions: {
          protobufField: {
            name: "optional_string_value",
            typeFullName: "string",
          },
        },
      }),
    }),
    extensions: {
      protobufMessage: {
        fullName: "testapis.proto3_optional.Message",
        name: "Message",
        package: "testapis.proto3_optional",
      },
    },
  });
"
`;
